src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs

```diff
@@ -371,9 +371,31 @@ private void UpdateInputScope(InputScope inputScope)
 
 		partial void OnTextWrappingChangedPartial(DependencyPropertyChangedEventArgs e)
 		{
-			//TODO : see bug #8178
+			if (e.NewValue is TextWrapping textWrapping && !string.IsNullOrEmpty(Text) &&
+				(textWrapping is TextWrapping.Wrap || textWrapping is TextWrapping.WrapWholeWords))
+			{
+
+				if (_textBoxView != null)
+				{
+					_isSingleLine = _textBoxView.IsSingleLine;
+
+					if (_textBoxView.IsSingleLine)
+					{
+						_textBoxView.SetSingleLine(false);
+					}
```

Comment:
Maybe this needs comments to explain the pattern.  I don't understand what it is doing.

My questions remains: why do we need such `_isSingleLine` variable? What does the null value means here?

Also... If I trigger wrap/non-wrap on the TextBox, I'm not sure it will reset the `_isSingleLine` to the correct value.

Reply:
As I said, needs to comment the pattern and explain:
1. why we need to keep the value of `_isSingleLine` in a field
2. what the `null` value mean on this field.
3. ensure the reset is done properly when the control is reused.

Reply:
Questions not answered